unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WalletHUD/WalletSectionHUDController.cs

```diff
@@ -0,0 +1,176 @@
+            dataStore.wallet.currentEthereumManaBalance.OnChange -= OnCurrentEthereumManaBalanceChanged;
+            dataStore.wallet.currentPolygonManaBalance.OnChange -= OnCurrentPolygonManaBalanceChanged;
+            dataStore.wallet.isWalletSectionVisible.OnChange -= OnWalletSectionVisible;
+            ownUserProfile.OnUpdate -= OnProfileUpdated;
+            view.OnBuyManaClicked -= GoToManaPurchaseUrl;
+            view.OnLearnMoreClicked -= GoToLearnMoreUrl;
+
+            fetchEthereumManaCancellationToken.SafeCancelAndDispose();
+            fetchPolygonManaCancellationToken.SafeCancelAndDispose();
+        }
+
+        private void ConfigureWalletSectionInFullscreenMenuChanged(Transform currentParentTransform, Transform _) =>
+            view.SetAsFullScreenMenuMode(currentParentTransform);
+
+        private void OnCurrentEthereumManaBalanceChanged(double currentBalance, double _) =>
+            view.SetEthereumManaBalance(currentBalance);
+
+        private void OnCurrentPolygonManaBalanceChanged(double currentBalance, double _) =>
+            view.SetPolygonManaBalance(currentBalance);
+
+        private void OnWalletSectionVisible(bool isVisible, bool _)
+        {
+            if (ownUserProfile.isGuest)
+                return;
+
+            if (isVisible)
+            {
+                fetchEthereumManaCancellationToken = fetchEthereumManaCancellationToken.SafeRestart();
+                fetchPolygonManaCancellationToken = fetchPolygonManaCancellationToken.SafeRestart();
+                RequestEthereumManaAsync(fetchEthereumManaCancellationToken.Token).Forget();
+                RequestPolygonManaAsync(fetchPolygonManaCancellationToken.Token).Forget();
+            }
+            else
+            {
+                fetchEthereumManaCancellationToken.SafeCancelAndDispose();
+                fetchPolygonManaCancellationToken.SafeCancelAndDispose();
+            }
+        }
+
+        private async UniTask RequestEthereumManaAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                await UniTask.WaitUntil(() =>
+                    ownUserProfile != null && !string.IsNullOrEmpty(ownUserProfile.userId),
+                    cancellationToken: cancellationToken);
+
+                view.SetEthereumManaLoadingActive(true);
+                Promise<double> promise = theGraph.QueryEthereumMana(ownUserProfile.userId);
+                if (promise != null)
+                {
+                    await promise;
+                    dataStore.wallet.currentEthereumManaBalance.Set(promise.value);
+                    view.SetEthereumManaLoadingActive(false);
+                }
+
+                await UniTask.Delay(TimeSpan.FromSeconds(WalletUtils.FETCH_MANA_INTERVAL), cancellationToken: cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+            }
+        }
```

Comment:
same question about while loop. Also, I see this function duplicated already 4 times with the tiny diffirences? Would it be possible to generalize it?

Reply:
Here we have the same case that the explained before but for the case when we have the Wallet section open. I don't see the possibility of having an infinite loop except for the cases where this section is open, in which case it is correct to have an "infinite" loop.